osgar/drivers/lora.py

```diff
@@ -0,0 +1,82 @@
+"""
+  LoRa - Long Range Radio
+
+  Current protocol uses numbers for DeviceID (1, 2, ... 5) for source identification and '|' for data separation.
+  The whole packet is terminated by '\n'
+"""
+from datetime import timedelta
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+ALIVE_MESSAGE = b'alive'
+
+
+def parse_lora_packet(packet):
+    s = packet.split(b'|')
```

Comment:
So the return value of this function is (address, data)? Who's address?

Reply:
So the data is actually not binary but text? The IDs are one char? So an example message would be `123|....` and the meaning would be that devices `1`, `2` and `3` have already seen the data?

Reply:
At the moment ID is one char, and yes there could be more optimization available if necessary. Now the transmitter is "simple", does not parse anything, just adds <device ID> and <separator>. We may have more sophisticated protocol if this technology turns out to be valuable in SubT or other context.

Reply:
And yes, at the moment it is meant to be text messages (easier to debug, display on the radio screens)

Reply:
https://robotika.cz/articles/reviews/lora/lora-for-robots.jpg

Reply:
I am not arguing one is better than the other. I am just trying to understand. The comment you added says "parse block of binary data" but it is actually expecting textual string on the input. That is what confused me.

Reply:
OK - I wrote "binary" as this have to be bytes arrays ... I got your point.

Reply:
I would not write "binary data" as a reference to ascii data in bytes array but whatever.